### DISCUSSION TITLE
Icons for call direction

### DISCUSSION THREAD

**Discussion by yongjie1** [twilio-labs/paste#2134] [Request: Icon] [answered]
We'd like to add 3 icons to paste to indicate the call directions: Inbound, Outbound, Unknown.
Here are the link to streamline app:
Outbound https://app.streamlinehq.com/icons/streamline-light/phones-mobile-devices/phone-actions/phone-actions-call
Inbound https://app.streamlinehq.com/icons/streamline-light/phones-mobile-devices/phone-actions/phone-actions-receive
Unknown https://app.streamlinehq.com/icons/streamline-light/phones-mobile-devices/phone-actions/phone-actions-question
Design 
![image](https://user-images.githubusercontent.com/91665462/149339874-bc46324b-3589-4299-b92d-eba5775e926e.png)

**Comment by kate-porter** [comment]
Hey @yongjie1! Thanks for requesting these - these make sense to get added to Paste, so I'll create tickets to do so. When is this page going into implementation?

**Reply by yongjie1** [reply]
Thanks @katieporter. We aim to build and launch it in Q1 2022.

**Reply by kate-porter** [reply]
Got it, thanks! We'll aim to have these added to Paste within the next month if that works for you.

**Reply by yongjie1** [reply]
Great, thanks!

**Comment by loreina** [comment]
Hi @yongjie1, just wanted to follow up here. Could you share more about the use case for an Unknown Call icon? We're wondering how this icon would be used in context with the inbound and outbound call icons. If you have screenshots to share, that would be amazing thanks!

**Reply by yongjie1** [reply]
Hi @loreina, **Unknown Calls** are calls that we don't know where the direction of it in the backend and they are not belongs to Inbound, Outbound, or Internal call, we'll  show it as "Unknown". We get these data from customers and then we show it in the table that I've shared in the post.

The team has decided to add another call direction: Internal Call (calls from internal agent to internal agent). I can't find any good icons for internal call from Streamline. Do you have any recommendation? I was thinking about something like this 
![image](https://user-images.githubusercontent.com/91665462/153245491-d7b20db7-f399-4ba8-933d-ddea3c1eef69.png)

**Comment by loreina** [comment]
Hi @yongjie1, the team looked over the Figma file you shared and have landed on a couple of things:

- We'll add the Inbound Call and Outbound Call icons to Paste for inbound and outbound calls.
- We'll add a general Call icon that would work for unknown calls, and can be generalized for other use cases that may need a call icon.
- We suggest using the Agent icon for internal calls between agents. In Figma, it's called `other/Agent` icon.
- We also suggest adding description text next to the icon in the content cell to indicate to a user when a call is outbound, inbound, unknown, or internal. Adding description text provides the user with more context into the direction beyond using an icon. There's a variant of Table in Figma with Icon + description text that would work here.

I've updated the ticket in our backlog to add these icons and we'll follow up here when those have been added. Let me know if you have any questions, thanks!

**Reply by yongjie1** [reply]
@loreina Thanks for the update.
- For general call icon, do you mean something like this? Is there any reason that the team don't add the **phone call with ?** icon?
![image](https://user-images.githubusercontent.com/91665462/153648420-7555cfed-6df2-4c08-aa78-462e2bca80f2.png)
- The reason that we use icons in the table is to save space for other data. If we have icon + text, it'll occupy more space. Could we use tooltip? When users hover over the icon, they see some contextual info. 
![image](https://user-images.githubusercontent.com/91665462/153649287-52251ce6-5fbb-493b-89a3-5bfcf4612830.png)

**Reply by loreina** [reply]
We'd like to add a general call icon instead of an unknown call icon as its very use case specific so a dedicated icon doesn't feel quite right, especially when the question mark could have various interpretations.

We would also strongly recommend pairing the icon with description text rather than using a tooltip as it isn't immediately clear to a user what the icon could refer to and this would require the user to hover on the icon to understand its semantic meaning. Displaying the call direction in the table provides immediate guidance and contextual info on what it means without the user having to look at the arrow direction or hover on the icon.

Hope this clarifies things, let me know if I can help with anything else!

**Reply by yongjie1** [reply]
@loreina I have discussed with my team, we have a question about the agent icon for Internal call. Where in Twilio do we use this icon and what does it mean? Just want to make sure that there isn't any icon conflict across Twilio products - same icon has different meanings.
<img width="78" alt="image" src="https://user-images.githubusercontent.com/91665462/154294035-24c44b5f-d520-4a5b-8d84-2a9dad9beced.png">
 
The team have decided to go with icon + tooltip because we have lots of data in the table, we'd like to save some space for more important info. Moreover, as an agent who will use the tool on a day to day basis for work, they may has a little learning curve to understand the meaning of the 4 icons at the beginning but once they've learned it, they'll know the meanings without hovering over the icons.

To make it easier for agents to identify the 4 icons, we'd like to suggest a slightly different icon for inbound call direction. See the screen, **Inbound** icon mirrors the **Outbound** icon. 
<img width="209" alt="image" src="https://user-images.githubusercontent.com/91665462/154300136-54a7764e-724c-4f0f-b7d9-ea305deedaeb.png">

**Reply by loreina** [reply]
The Agent icon is used in places like SendGrid support, Flex interactions timeline, and transcription logs. We recommend using this icon for **internal calls** in this use case, where the call is between agent to agent. We'll add a general call icon for **unknown calls**.

We strongly advise against using a Tooltip to indicate the call direction to users. Just displaying the icon without description text doesn't feel like it's immediately evident to a user what the direction is. We'd recommend testing this implementation with users to prove that this method to display inbound and outbound icon's semantic meaning is clear and easily discoverable.

Thanks for the suggestion! We'll consider the mirroring when working on the ticket.

**Reply by yongjie1** [reply]
@loreina What does the icon mean in the other products? Does it mean "Agent" or does it also have other meanings?

We've seen many products that using icon with tooltip and we think people already build up the mental model of seeing an unfamiliar icon and hovering over it to understand what it means. E.g In Gmail, they use a star icon without text but if you hover over it you see the tooltip. In Google doc or Github comment, they have lots of icons and user needs to hover to understand the meaning
<img width="229" alt="image" src="https://user-images.githubusercontent.com/91665462/154328067-f0895368-3069-407c-b45a-7155b5aeb6ff.png">
<img width="483" alt="image" src="https://user-images.githubusercontent.com/91665462/154328118-03995b3e-3151-4e0e-9b08-00e1d1d8e0a3.png">
<img width="525" alt="image" src="https://user-images.githubusercontent.com/91665462/154328194-0e6ea31a-b430-4b5e-be58-4b0cff27c02e.png">
<img width="924" alt="image" src="https://user-images.githubusercontent.com/91665462/154328443-41561884-193c-4ed3-9367-741c7acb5351.png">

**Reply by loreina** [reply]
Yeah, the icon means "Agent" and is meant to be used in products to indicate as such.

The examples you shared are icon buttons, so they're interactive and the tooltip is displaying text on how that element works. A Tooltip should only be placed on a focusable element (like the ones in the screenshots above) so that it's accessible for screen readers. It shouldn't be used when hovering over an icon as screen readers would not have access to the text within the Tooltip. We have more information on this on our doc site: https://paste.twilio.design/components/tooltip/#accessibility

**Reply by yongjie1** [reply]
@loreina Thanks, I see. I understand it's good to have icon with text but in our case we really don't have enough space and this is not the most important info to users. We are thinking of using the info icon in the column header to explain what each icon means. It is a typical pattern we already use to explain what each column is about and the discoverability should be alright. As it's only 4 icons, it's okay to have a little learning curve. Once they've seen the meanings, the leaning curve is pretty low. Especailly when they are using the product on a day to day basis.
<img width="608" alt="image" src="https://user-images.githubusercontent.com/91665462/154521773-f2724160-7922-4f7d-90fe-df7cf8636d59.png">

**Some more questions about the icons:**
If the agent icon means agent in the other products, won't it be confusing to our customers if they see it means "Internal call" in our product?
I've done a quick test with our team on the icons and here are the feedback on the meaning of the icons. Inbound & Outbound are pretty easy to understand however the Agent icon and the Phone icon do not mean "Internal call" and "Unknown call" to them. 

They are also unclear about the "Agent" icon and the "Unknown" call direction icon...

**Feedback**
<img width="912" alt="image" src="https://user-images.githubusercontent.com/91665462/154523247-60f70dfa-ba1e-4340-aebe-0d71b345b4ba.png">
<img width="595" alt="image" src="https://user-images.githubusercontent.com/91665462/154523533-b50debaf-7bd0-49d7-b6f7-ff1b697f7f35.png">

**Reply by loreina** [reply]
I spoke with the team about this and we would still recommend pairing the icon with text for the call direction.

I've also mocked up a few ideas in this [Figma file](https://www.figma.com/file/SpeLpBoqN2vZv8KWcSn0J1/?node-id=1103%3A85059) where some of these ideas don't make the cell wider:
1. **Icon + Text**: This uses the table cell content for Icon with description text. Icons should be paired with text to be accessible for screen readers. I am using the `voiceCapable` icon as a placeholder and that shouldn't be used in the final mockups; we'll have the call direction icons for production use.
2. **Text only**: This displays the call direction without any icon. The current list of call direction would be around the same text length as the word "Direction" in the header, that sets the width of that column.
3. **Badge w/ Icon**: We're using the decorative Badges here, but with a prefix icon. The width of these badges are longer than the "Direction" header text so may not be as useful if you're trying to save table space.
4. **Badge**: We're also using the decorative Badges here. I've used each of the 4 available decorative variants for each call direction. It takes up about the same amount of width space as "Direction" does in the header. You can see a live example of this on our doc site: https://paste.twilio.design/components/badge/#call-attention-to-attributes-of-table-items. This would be our preferred approach.

Regarding the Agent icon, it is used to mean "agent" and in this use case, would be a worthwhile learning curve for users to understand that it indicates an "internal call", or a call between agents. 

Regarding the unknown call and internal call icons, it's great that you got feedback from your team. Have you done any external user testing or research on these icons? If we add an icon that we're not confident about, it's very hard to remove it and we should be wearing the customers' shoes to make those decisions. 

Hope this is helpful! Please come to office hours if you have any more questions and we can discuss in-person.

**Reply by yongjie1** [reply]
@loreina thanks, I've discussed it with my team and we've decided to go with text for now. We'll pause the request for icons.

**Comment by loreina** [comment, marked as answer]
@yongjie1 Great, thanks! Happy we had this discussion and landed on an option that's accessible. I'll close the ticket in our backlog for the icons. Feel free to create a new discussion for future requests. Thanks again!
